aspnetcore/fundamentals/static-files.md

```diff
@@ -4,68 +4,281 @@ author: rick-anderson
+
+The call `app.UseFileServer` enables the serving of static files and the default file. Directory browsing isn't enabled. The following code shows `Startup.Configure` with `UseFileServer`:
+
+[!code-csharp[](~/fundamentals/static-files/samples/3x/sample/StartupEmpty2.cs?name=snippet&highlight=14)]
+
+The following code enables default documents and directory browsing:
+
+```csharp
+app.UseFileServer(enableDirectoryBrowsing: true);
+```
+
+The following code shows `Startup.Configure` with the preceding code:
+
+[!code-csharp[](~/fundamentals/static-files/samples/3x/sample/StartupEmpty3.cs?name=snippet&highlight=14)]
+
+Consider the following directory hierarchy:
+
+* `wwwroot`
+  * `css`
+  * `images`
+  * `js`
+* `MyStaticFiles`
+  * `images`
+    * `MyImage.jpg`
+  * `default.html`
+
+The following code enables static files, default files, and directory browsing of `MyStaticFiles`:
+
+[!code-csharp[](~/fundamentals/static-files/samples/3x/sample/StartupUseFileServer.cs?name=snippet&highlight=4,20-30)]
+
+<xref:Microsoft.Extensions.DependencyInjection.DirectoryBrowserServiceExtensions.AddDirectoryBrowser%2A> must be called when the `EnableDirectoryBrowsing` property value is `true`.
 
-::: moniker range="< aspnetcore-2.0"
+Using the file hierarchy and preceding code, URLs resolve as follows:
+
+| URI            |      Response  |
+| ------- | ------|
+| `https://<hostname>/StaticFiles/images/MyImage.jpg` | *MyStaticFiles/images/MyImage.jpg* |
+| `https://<hostname>/StaticFiles` | *MyStaticFiles/default.html* |
+
+If no default-named file exists in the *MyStaticFiles* directory, `https://<hostname>/StaticFiles` returns the directory listing with clickable links:
+
+![Static files list](static-files/_static/db2.png)
+
+<xref:Microsoft.AspNetCore.Builder.DefaultFilesExtensions.UseDefaultFiles*> and <xref:Microsoft.AspNetCore.Builder.DirectoryBrowserExtensions.UseDirectoryBrowser*> perform a client-side redirect from the target URI without a trailing `/`  to the target URI with a trailing `/`. For example, from `https://<hostname>/StaticFiles` to `https://<hostname>/StaticFiles/`. Relative URLs within the *StaticFiles* directory are invalid without a trailing slash (`/`).
+
+## FileExtensionContentTypeProvider
+
+The [FileExtensionContentTypeProvider](/dotnet/api/microsoft.aspnetcore.staticfiles.fileextensioncontenttypeprovider) class contains a `Mappings` property serving as a mapping of file extensions to MIME content types. In the following sample, several file extensions are registered to known MIME types. The *.rtf* extension is replaced, and *.mp4* is removed:
+
+[!code-csharp[](~/fundamentals/static-files/samples/3x/sample/StartupFileExtensionContentTypeProvider.cs?name=snippet2)]
+
+The following code shows `Startup.Configure` with the preceding code:
+
+[!code-csharp[](~/fundamentals/static-files/samples/3x/sample/StartupFileExtensionContentTypeProvider.cs?name=snippet)]
```

Comment:
```suggestion
[!code-csharp[](~/fundamentals/static-files/samples/3x/sample/StartupFileExtensionContentTypeProvider.cs?name=snippet&highlight=14-42)]
```

This might be a bit long for a highlight; I'm suggesting it just for consitency.